content/ADR-239-avatar-assembly-changes.md

```diff
@@ -0,0 +1,197 @@
+
+The following example illustrates the expected behavior: If a wearable of type "Mask" hides another wearable of type "Hat", and this in turn hides wearables of type "Mask", the wearable of type "Mask" will have visualization priority, hiding the Mask.
+
+
+### Hiding and replacing
+
+The current result of equipping an item that replaces or hides other wearables is unexpected and inconsistent. When a player navigates through their inventory items, as they equip, unequip, or hide other wearables from other categories without notice, they cannot reverse this behavior when equipping a different wearable.
+When a wearable unequips another wearable, the latter could be running a portable experience. By deactivating it, any progress or data in memory could be lost, ruining its expected behavior.
+The decision-making capacity of creators is overlapping with the freedom of choice of players. When a player purchases a wearable, thinking it will combine well with what they already have in their inventory, they may be unpleasantly surprised that these are not compatible. Information about which other wearables are replaced or hidden is not clearly indicated in the marketplace or backpack.
+There is no real weighty reason for a wearable to unequip another wearable, instead of hiding it to prevent clipping problems.
+
+On the base of that the replacing functionality for new wearables will be removed, pre-existing wearables will see the replacing list being merged with the hiding list.
+
+### Body fallback
+
+Previously, by unequipping the upper_body, the lower_body, the feet or the eyebrows, caused the system to fallback to default wearables (t-shirt, pants and shoes).
+This prevents, for example, players from going barefoot or without t-shirt.
+
+The system will extent the concept of Body by removing the default fallback wearables.
+These pieces, which can be called "Bare feet", "Naked Top" and "Naked Bottom", are not considered wearables, but rather parts of the base body geometry that are exclusively displayed when no wearables are equipped and there are no other wearables hiding the relevant category.
+
+The base body geometry and textures will change to make them more suitable for an environment for all audiences.
+
+### Force Render
+
+To extend further players ability to use the wearables they have in all their glory. If an equipped wearable is hidden by the effect of another wearable, players can unhide them and override this instruction.
+This forced render effect is removed as soon as another wearable that re-hides that category is equipped, forcing the user to show it again.
+
+In order to store the force render categories, it is proposed to add a new field to the player profile holding a list of overridden hide categories.
+This list of string will hold the current overridden categories so that each player can render other players (and itself) according to the new rules.
+As an example:
+```
+"forceRender": ["mask","top_head","feet"]
```

Comment:
Maybe, if we are still on-time, forcedCategories would be a better self-explanatory name
```suggestion
"forcedCategories": ["mask","top_head","feet"]
```

Reply:
sadly too late

Reply:
the functionality has already been added to unity renderer, browser interface and profile deploy validation.
Also I think that forceRender is descriptive enough, considering that something hidden is still equipped, so by adding it to the list we are forcing it to render even if hidden